newIDE/app/src/ProjectsStorage/LocalFileStorageProvider/LocalFileResourcesWatcher.js

```diff
@@ -36,28 +37,34 @@ export const setupResourcesWatcher =
         const folderPath = path.dirname(fileMetadata.fileIdentifier);
         const gameFile = path.basename(fileMetadata.fileIdentifier);
         const autosaveFile = gameFile + '.autosave';
-        const watcher = fileWatcher
-          .watch(folderPath, {
-            ignored: [
-              `**/.DS_Store`,
+        ipcRenderer.on('project-file-changed', (event, path) => {
+          // TODO: Is it safe to let it like that since the OS could for some reason
+          // do never-ending operations on the folder or its children, making the debounce
+          // never ending.
+          debouncedCallback(path);
+        });
+        const subscriptionIdPromise = ipcRenderer.invoke(
+          'local-filesystem-watcher-setup',
+
+          folderPath,
```

Comment:
random thought. Any reason we're watching **everything** instead of only looking at the resources registered on the project & updating the resourceWatcher accordingly?
(this may not be any more efficient if this filters after watching)

Reply:
With parcel watcher, you can only watch a folder so we don't have much choice 😅 .
But with this library, filtering doesn't seem to be an issue.